### PR TITLE
AlertFilters: Use Log4j 2.x

### PR DESCRIPTION
**Pull request by kingthorin**
- Update build file, CHANGELOG, and change code to use updated logging infrastructure.

Part of: zaproxy/zaproxy#6376

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFilter.java [hunk 1]
```diff
@@ -343,27 +342,12 @@ private static boolean matchesStringOrRegex(
...
-                        log.debug(
-                                "Filter didnt match "
-                                        + paramName
-                                        + " regex: "
-                                        + paramValue
-                                        + " : "
-                                        + targetValue);
-                    }
+                    log.debug(
+                            "Filter didn't match {} : {} : {}", paramName, paramValue, targetValue);
```
Comment (ricekot) [inline addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFilter.java:346]: Was "regex: " omitted on purpose?

Comment (kingthorin) [reply addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFilter.java:346]: Nope, thanks for catching that.

**Comment by kingthorin** [comment]
Fixed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Pull request merged**